### DISCUSSION TITLE
Comma separated list seems to not work on scopes

### DISCUSSION THREAD

**Discussion by K2ouMais** [spatie/laravel-query-builder#857] [Q&A] [answered]
Hello everyone,

I am using a route like this:

```/api/user?filter[userId]=1,2,3```

In the controller I use a scope to query the id.

```php
AllowedFilter::scope('userId')
```

The scope in the model:

```php
public function scopeUserId(Builder $query, $id): Builder
{
    dd($id); // output: 1
    return $query->where('id', $id);
}
```

The output of ```dd($id)``` seems to only be the first value of the comma separated list.

Is this correct? What am I doing wrong? I am expecting it to be ```1,2,3```.

Thank you in advance.

**Comment by K2ouMais** [comment]
@freekmurze Thank you for putting this into Discussions, but I have already posted it here and never got an answer.

It would be really nice if you could say, that this is the wanted behavior.

**Comment by dkstudio86** [comment, marked as answer]
Maybe is late... But for references...

```
public function scopeUserId(Builder $query, string ...$id): Builder
{
    dd($id); // output: array(1, 2, 3)
    return $query->whereIn('id', $id);
}

```
The '...' make the magic.

Reference:
https://www.php.net/manual/en/functions.arguments.php#functions.variable-arg-list

**Reply by K2ouMais** [reply]
I cant test this right now, but why is it beeing passed as an argument list?

It should be mentioned in the docs, so that people dont ask the same as me.

Thank you for your help
